pkg/apis/hive/v1alpha1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -250,11 +246,15 @@ func checkValidPatchTypes(patches []hivev1.SyncObjectPatch) (string, bool) {
 	return "", true
 }
 
-func checkSecretReferences(secrets []hivev1.SecretReference) (string, bool) {
+func checkSecretReferences(secrets []hivev1.SecretReference) field.ErrorList {
+	errs := field.ErrorList{}
 	for _, secret := range secrets {
+		if len(secret.Source.Kind) == 0 {
+			errs = append(errs, field.Required(field.NewPath("kind"), "SecretReference.Source.Kind is required"))
+		}
 		if secret.Source.Kind != "secret" {
-			return secret.Source.Kind, false
+			errs = append(errs, field.Invalid(field.NewPath("kind"), secret.Source.Kind, fmt.Sprintf("Invalid SecretReference.Source kind detected: %s. SecretReferences may only contain references to kind: secret", secret.Source.Kind)))
```

Comment:
This should be NotSupported instead of Invalid.

`field.NotSupported(fldPath.Index(i).Child("source").Child("kind"), secret.Source.Kind, []string{"secret"})`

Although, really, you need to check the entire GVK not just the kind. There could be another API group that has a kind of secret.